rets/http/parsers/parse.py

```diff
@@ -16,7 +16,18 @@
 
 def parse_xml(response: ResponseLike) -> etree.Element:
     encoding = response.encoding or DEFAULT_ENCODING
-    root = etree.fromstring(response.content.decode(encoding), parser=etree.XMLParser(recover=True))
+    try:
+        root = etree.fromstring(response.content.decode(encoding), parser=etree.XMLParser(recover=True))
+    except ValueError as e:
+        if str(e) == "Unicode strings with encoding declaration are not supported. Please use bytes input or XML fragments without declaration.":
```

Comment:
Is there a code or metadata attached that allows us to avoid exact string matching? Worried it might break everything unexpectedly if we upgrade dependencies in the future.

Reply:
unfortunately not; this is the source code:
https://github.com/lxml/lxml/blob/ea954da3c87bd8f6874f6bf4203e2ef5269ea383/src/lxml/parser.pxi#L1892